source/libs/wal/src/walWrite.c

```diff
@@ -252,23 +252,36 @@ static FORCE_INLINE int32_t walCheckAndRoll(SWal *pWal) {
     }
   }
 
+  if (walGetLastFileCachedSize(pWal) > WAL_RECOV_SIZE_LIMIT / 2) {
```

Comment:
If the size of data not-synced yet since last time is larger than a threshold, save them with fsync.